sklearn/model_selection/tests/test_get_params_masked_array.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import pytest
+
+from sklearn.model_selection._search import _get_param_masked_arrays
+from sklearn.preprocessing import (
+    OneHotEncoder,
+    OrdinalEncoder,
+)
+
+# Construct these outside the tests so that the same object is used
+# for both input and `expected`
+one_hot_encoder = OneHotEncoder()
+ordinal_encoder = OrdinalEncoder()
+
+# If we construct this directly via `MaskedArray`, the list of tuples
+# gets auto-converted to a 2D array.
+ma = np.ma.MaskedArray(np.empty(2), mask=True, dtype=object)
```

Comment:
It looks like something like this works:
```py
my_iter = iter([(1, 2), (3, 4)])
masked_array = np.fromiter(my_iter, dtype=object)
```

Reply:
...but not in old versions of numpy 😭